### PR TITLE
19453 added legalName field to draft_results to link with auth-api

### PR DESCRIPTION
**Pull request by PaulGarewal**
*Issue #:* /bcgov/entity#19453

*Description of changes:*

- added legalName field to draft_results to link with auth-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by argush3** [comment]
@kialj876 added you as I think you might use this endpoint too.

**Comment by argush3** [comment]
Can we update some tests to verify this is working as expected?

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 1]
```diff
@@ -173,7 +173,11 @@ def search_businesses():
             {
                 'identifier': x.temp_reg,
                 'legalType': x.json_legal_type,
-                **({'nrNumber': x.json_nr} if x.json_nr else {})
+                **({'nrNumber': x.json_nr} if x.json_nr else {}),
+                **({'legalName': x.filing_json.get('filing', {})
+                    .get(x.filing_type).get('nameRequest', {})
+                    .get('legalName')}
+                   if x.filing_type == 'amalgamationApplication' else {})
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/business/business.py:180]: I think we don't need this to be specific to amalgamation

File: legal-api/tests/unit/resources/v2/test_business.py [hunk 2]
```diff
@@ -399,6 +400,16 @@ def test_post_affiliated_businesses(session, client, jwt):
...
+    # verify 'legalName' for each draft entity
+    for draft_entity in rv.json['draftEntities']:
+        identifier = draft_entity['identifier']
+        expected_draft_business = next((draftb for draftb in draft_businesses if draftb[0] == identifier), None)
+        if expected_draft_business and expected_draft_business[3]:
+            # if NR number is present, assert 'legalName' is also expected to be present
+            assert 'legalName' in draft_entity
+        else:
+            # assert 'legalName' is None or empty if no NR number is provided
+            assert draft_entity.get('legalName') is None or draft_entity.get('legalName') == ''
```
Comment (PaulGarewal) [inline legal-api/tests/unit/resources/v2/test_business.py:413]: Not sure if this is too thorough for testing.

File: legal-api/tests/unit/resources/v2/test_business.py [hunk 3]
```diff
@@ -376,7 +376,8 @@ def test_post_affiliated_businesses(session, client, jwt):
         json_data['filing']['header']['legalType'] = draft_business[2]
         if draft_business[3]:
             json_data['filing'][filing_name] = {
-                'nameRequest': {'nrNumber': draft_business[3]}
+                'nameRequest': {'nrNumber': draft_business[3],
+                                'legalName': 'name example'}
```
Comment (PaulGarewal) [inline legal-api/tests/unit/resources/v2/test_business.py:380]: Seemed like the correct spot to place 'legalName', but unsure.

File: legal-api/src/legal_api/resources/v2/business/business.py [hunk 4]
```diff
@@ -173,7 +173,10 @@ def search_businesses():
             {
                 'identifier': x.temp_reg,
                 'legalType': x.json_legal_type,
-                **({'nrNumber': x.json_nr} if x.json_nr else {})
+                **({'nrNumber': x.json_nr} if x.json_nr else {}),
+                **({'legalName': x.filing_json.get('filing', {})
+                    .get('nameRequest', {})
+                    .get('legalName')})
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/business/business.py:179]:
```suggestion
                **({'legalName': x.filing_json.get('filing', {})
                    .get(x.filing_type, {})
                    .get('nameRequest', {})
                    .get('legalName')})
```

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Pull request merged**